### PR TITLE
SWIFT-355 Make option properties mutable

### PR DESCRIPTION
**Pull request by ncarbon**
Makes all option properties mutable. Option properties are declared with 'var' instead of 'let'.

### CODE REVIEW

File: Sources/MongoSwift/MongoCollection+Read.swift [hunk 1]
```diff
@@ -377,8 +404,6 @@ public struct FindOptions: Codable {
         self.comment = comment
         // although this does not get encoded, we store it for debugging purposes
```
Comment (patrickfreed) [inline Sources/MongoSwift/MongoCollection+Read.swift]: [nit] this comment can be removed.

File: Sources/MongoSwift/MongoCollection+Read.swift [hunk 2]
```diff
@@ -278,67 +278,67 @@ public enum CursorType {
...
     /// Specifies a collation.
-    public let collation: Document?
+    public var collation: Document?
 
     /// Attaches a comment to the query.
-    public let comment: String?
+    public var comment: String?
 
     /// If a `CursorType` is provided, indicates whether it is `.tailable` or .`tailableAwait`.
-    private let tailable: Bool?
```
[the remark this answers is not in the document]
Comment (patrickfreed) [reply Sources/MongoSwift/MongoCollection+Read.swift]: I think we can also leave `tailable` and `await` as `let` since we don't want to be modifying them directly but only through the new cursor type property.

[the remark this answers is not in the document]
Comment (kmahar) [reply Sources/MongoSwift/MongoCollection+Read.swift]: If we declare them with `let`, how would we modify them as needed in my example above?

File: Tests/MongoSwiftTests/MongoCollectionTests.swift [hunk 3]
```diff
@@ -307,17 +307,31 @@ final class MongoCollectionTests: MongoSwiftTestCase {
...
+        expect(try encoder.encode(tailable)).to(equal(["awaitData": false, "tailable": false ]))
+
+        var tailableAwait = FindOptions(cursorType: .tailableAwait)
         expect(try encoder.encode(tailableAwait)).to(equal(["awaitData": true, "tailable": true ]))
+
+        tailableAwait.cursorType = .tailable
+        expect(try encoder.encode(tailableAwait)).to(equal(["awaitData": false, "tailable": true ]))
+
+        // test nill cursorType
+        var nilTailable = FindOptions(cursorType: nil)
```
Comment (kmahar) [inline Tests/MongoSwiftTests/MongoCollectionTests.swift]: since this doesn't get modified, you can make it a `let` constant. (the compiler will generate a warning about this not needing to be mutable every time we build the tests.)

however, it might be worth keeping it mutable, and adding a few more lines to this test to also check what happens when `cursorType` is initially `nil` and is later set to something. (and maybe vice versa, where the user initially provided a cursorType and then the user sets it to nil.)
